### PR TITLE
added support for Seek on iterator and IteratorWithRange on storage

### PR DESCRIPTION
**Pull request by jbpin**
As discuss in issue #82 Here is the pull request. 

I write light test on memory.. I'm looking forward for your code review.

### CODE REVIEW

File: iterator.go [hunk 1]
```diff
@@ -9,6 +9,7 @@ type Iterator interface {
 	Key() string
 	Value() (interface{}, error)
 	Release()
+	Seek(key []byte) bool
```
Comment (db7) [inline iterator.go]: `goka.Iterator` has keys as strings. Does `Seek()` make sense when based on strings?

**Comment by jbpin** [comment]
Hi guys, any update on this ?

**Comment by jbpin** [comment]
Sure ;)

```
iterator.Seek([]byte('this is my key'))
```

**Comment by db7** [comment]
;)

What I meant is that the whole interface of `goka.Iterator` is defined with keys as strings. So `Seek()` should take a string, like this:

```
type Iterator interface {
	Next() bool
	Key() string
	Value() (interface{}, error)
	Release()
	Seek(key string) bool
}
```

**Comment by jbpin** [comment]
;) I see... In fact I added like that so that it's complient with the interface of the leveldb library. 
https://github.com/syndtr/goleveldb/blob/master/leveldb/iterator/iter.go

**Comment by db7** [comment]
I guessed so. The `storage.Iterator` is defined with keys as byte slices because LevelDB uses byte slices.

Still, since we work with string keys at the goka package level, I think it would make sense to keep string keys there. Or change all keys to byte slices, but that would break way too many things.

**Comment by jbpin** [comment]
ok get it.. I update the code

**Comment by db7** [comment]
Excellent! And this one should take strings too: `func (v *View) IteratorWithRange(start, limit []byte) (Iterator, error) {`

Besides these `[]byte` thing, the PR looks pretty good to me. Thanks for contributing.

**Comment by jbpin** [comment]
ok let me fix that

**Pull request merged**